topics/threaded-io.md

```diff
@@ -0,0 +1,119 @@
+# Threaded I/O
+## Introduction
+In order to keep it simple and reliable, Redis is mostly single-threaded. The implementation of parallel query execution might set an entry barrier for every new feature. However, there are certain threaded operations such as UNLINK, slow I/O accesses and other things that are performed on side threads.
+
+Now it is also possible to handle Redis clients' socket reads and writes in different I/O threads. Since especially writing is so slow, normally Redis users use pipelining in order to speed up the Redis performances per core, and spawn multiple instances in order to scale more. Using I/O threads it is possible to easily speedup two times Redis without resorting to pipelining nor sharding of the instance.
+
+## Configuration
+We have two options to control how I/O threads perform.
+```
+io-threads [thread-num]
+```
+The `io-threads` option control how many threads you want to use on reading/writing client connections.
+- 1 or not configured: Use single thread like what we have in the older version.
+- n (n >= 2): Use n threads(1 main thread & n-1 I/O threads) to handle network reads/writes.
+
+```
+io-threads-do-reads [yes|no]
```

Comment:
I would punt the configuration to the end or interleave the configuration with the introduction of the topics. io-threads-do-reads is hard to understand without context.